### PR TITLE
Adapt ingress to networking/v1

### PR DESCRIPTION
**Pull request by czomo**
Following ingress beta deprecation [link](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#ingress-v122) I switched it to networking/v1. networking/v1 is available in k8s from 1.19. Additionally added option to specify the ingressClassName. Please review
Fixes https://github.com/dragonflyoss/helm-charts/issues/68

### CODE REVIEW

**Comment by czomo** [comment]
@gaius-qi Can you take a look?

File: charts/dragonfly/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: dragonfly
 description: Dragonfly is an intelligent P2P based image and file distribution system
 icon: https://raw.githubusercontent.com/dragonflyoss/Dragonfly2/main/docs/images/logo/dragonfly.svg
 type: application
-version: 0.5.52
+version: 0.5.53
 appVersion: 0.5.52
```
Comment (yxxhero) [inline charts/dragonfly/Chart.yaml:7]: need update to 0.5.53

**Comment by yxxhero** [comment]
@czomo please fix lint error.

add docs:
```
networking.k8s.io/v1beta1 == 1.14 to 1.18

networking.k8s.io/v1 = 1.19+
```

File: charts/dragonfly/Chart.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ sources:
 
 annotations:
   artifacthub.io/changes: |
-    - Update dragonfly image version to v2.0.2
+    - Addapt ingress to networking/v1
```
Comment (jim3ma) [inline charts/dragonfly/Chart.yaml:29]: `Addapt` -> `Adapt`

**Review by jim3ma — CHANGES_REQUESTED** [review]

**Comment by jim3ma** [comment]
For generating helm docs, you need install `helm-docs` instead of editing by hand.

**Comment by gaius-qi** [comment]
@czomo Please fix lint workflows.

**Comment by czomo** [comment]
> @czomo please fix lint error.
> 
> add docs:
> 
> ```
> networking.k8s.io/v1beta1 == 1.14 to 1.18
> 
> networking.k8s.io/v1 = 1.19+
> ```
Linter fixed with helm docs, It changes some white spaces tho. Not sure if it case to bother.

Regarding additional docs would you like to mention that in readme/Chart.yaml or somewhere else? Please share. I don't think it is worth to adapt it to lover apiVersion as this version of k8s is no longer supported.

**Comment by czomo** [comment]
> Can we detect k8s version to do this ? refer: https://www.sobyte.net/post/2021-11/helm-chart-compatible-different-kube-versio/
Yes, sure. Take a look

File: charts/dragonfly/values.yaml [hunk 3]
```diff
@@ -941,10 +941,14 @@ manager:
   ingress:
     # -- Enable ingress
     enable: false
+    # -- Ingress class name
+    className: ""
```
Comment (jim3ma) [inline charts/dragonfly/values.yaml:945]: `className` and `pathType` need k8s 1.18+

Comment (czomo) [reply charts/dragonfly/values.yaml:945]: Note added to docs

Comment (czomo) [reply charts/dragonfly/values.yaml:945]: Detecting if we can add className is already implemented  [here](https://github.com/dragonflyoss/helm-charts/pull/71/files#diff-dff1127d01114f9648a56cc0d44794d52b7bf497c2786c33942cc13c2bb9f94cR23), pathType [here](https://github.com/dragonflyoss/helm-charts/pull/71/files#diff-dff1127d01114f9648a56cc0d44794d52b7bf497c2786c33942cc13c2bb9f94cR42) We only need to check if w have k8s >=1.18

Comment (czomo) [reply charts/dragonfly/values.yaml:945]: Could you elaborate what do you expect to be done here in broad spectrum?

**Review by jim3ma — APPROVED** [review]
LGTM

**Pull request merged**